File: server/src/main/java/lighting/server/settings/SettingsServiceImpl.java
package lighting.server.settings;

import java.io.IOException;

import org.springframework.stereotype.Component;

import lighting.server.IO.IOService;

@Component
// migrated
public class SettingsServiceImpl implements SettingsService {

	private final IOService iOService;

	public SettingsServiceImpl(final IOService iOService) {
		this.iOService = iOService;
	}

	public void saveSettingsToDisk(Settings settings) throws IOException {
		iOService.saveSettingsToDisk(settings);
	}

	public Settings getSettingsFromDisk() throws IOException {
		return iOService.getSettingsFromDisk();
	}
}

File: server/src/main/scala/lassi/server/settings/SettingsServiceImpl.scala
package lassi.server.settings

import lassi.server.io.IoService
import org.springframework.stereotype.Component

@Component
class SettingsServiceImpl(ioService: IoService) extends SettingsService {

  override def readSettings: Settings = {
    ioService.readSettings
  }

  override def writeSettings(settings: Settings): Unit = {
    ioService.writeSettings(settings)
  }
}
